pensjon-brevbaker/src/main/kotlin/no/nav/pensjon/brev/template/Operations.kt

```diff
@@ -27,38 +21,47 @@ sealed class UnaryOperation<In, out Out> : Operation() {
 
     object ToString : UnaryOperation<Any, String>() {
         override fun apply(input: Any): String = input.toString()
+        override fun stableHashCode(): Int = "UnaryOperation.ToString".hashCode()
     }
 
     object SizeOf : UnaryOperation<Collection<*>, Int>(){
         override fun apply(input: Collection<*>): Int = input.size
+        override fun stableHashCode(): Int = "UnaryOperation.SizeOf".hashCode()
     }
 
     object AbsoluteValue : UnaryOperation<Int, Int>(){
         override fun apply(input: Int): Int = input.absoluteValue
+        override fun stableHashCode(): Int = "UnaryOperation.AbsoluteValue".hashCode()
     }
 
     object AbsoluteValueKroner : UnaryOperation<Kroner, Kroner>(){
         override fun apply(input: Kroner): Kroner = Kroner(input.value.absoluteValue)
+        override fun stableHashCode(): Int = "UnaryOperation.AbsoluteValueKroner".hashCode()
     }
 
-    data class Select<In: Any, Out>(val selector: TemplateModelSelector<In, Out>) : UnaryOperation<In, Out>() {
+    class Select<In: Any, Out>(val selector: TemplateModelSelector<In, Out>) : UnaryOperation<In, Out>() {
         override fun apply(input: In): Out = selector.selector(input)
+        override fun stableHashCode(): Int = 31 * "UnaryOperation.Select".hashCode() + selector.stableHashCode()
     }
 
-    data class SafeCall<In: Any, Out>(val selector: TemplateModelSelector<In, Out>) : UnaryOperation<In?, Out?>() {
+    class SafeCall<In: Any, Out>(val selector: TemplateModelSelector<In, Out>) : UnaryOperation<In?, Out?>() {
         override fun apply(input: In?): Out? = input?.let { selector.selector(it) }
+        override fun stableHashCode(): Int = 31 * "UnaryOperation.SafeCall".hashCode() + selector.stableHashCode()
```

Comment:
31* er for å få entropy fra både første å andre hashcode? Hadde vært fint med en funksjon for det kanskje, den var litt magisk.

Reply:
Stemmer. Er basert på hvordan hashcode genereres av kotlin for data class. Grunnen til at jeg lot denne her ligge sånn som dette er fordi TemplateModelSelector interfacet er deklarert i modulen template-model-generator, mens StableHash interfacet er deklarert i brevbaker-modulen, så jeg får ikke deklarert at TemplateModelSelector extends StableHash :cry:

Reply:
Hadde vært nice om jeg kunne på et vis skrevet `StableHash by selector.stableHashCode()`